components/builder-api/src/server/resources/profile.rs

```diff
@@ -206,6 +217,142 @@ async fn revoke_access_token(req: HttpRequest,
     }
 }
 
+#[allow(clippy::needless_pass_by_value)]
+async fn set_license(req: HttpRequest,
+                     state: Data<AppState>,
+                     Json(payload): Json<LicensePayload>)
+                     -> HttpResponse {
+    let conn = match state.db.get_conn().map_err(Error::DbError) {
+        Ok(conn_ref) => conn_ref,
+        Err(err) => return err.into(),
+    };
+
+    match authorize_session(&req, None, None) {
+        Ok(_session) => {
+            let base_url = &state.config.api.license_validation_url;
+            let license_url = format!("{}?licenseId={}&version=2", base_url, payload.license_key);
+
+            let response =
+                match reqwest::blocking::Client::new().get(license_url)
+                                                      .header("Accept", "application/json")
+                                                      .send()
+                {
+                    Ok(resp) => resp,
+                    Err(err) => {
+                        return HttpResponse::InternalServerError().body(format!("License API \
+                                                                                 error: {}",
+                                                                                err))
+                    }
+                };
+
+            if !response.status().is_success() {
+                return HttpResponse::BadRequest().body("Invalid license key or failed to fetch \
```

Comment:
Do not return a generic error, as it can hide important details like license expiration or invalid license. Instead, return the actual response from the server using:

return HttpResponse::build(status).body(body);
or HttpResponse::with_body....